openassessment/workflow/models.py

```diff
@@ -364,11 +420,27 @@ def set_score(self, score):
                 'points_possible'.
 
         """
-        sub_api.set_score(
-            self.submission_uuid,
-            score["points_earned"],
-            score["points_possible"]
-        )
+        if not self.staff_score_exists():
+            sub_api.set_score(
+                self.submission_uuid,
+                score["points_earned"],
+                score["points_possible"]
+            )
+
+    def staff_score_exists(self):
+        """
+        Check if a staff score exists for this submission. Cached after first use.
```

Comment:
Ack, more outdated documentation. I had tried to include that, but it wasn't working and I wasn't sure it would be invalidated regularly enough for the risk of a stale cache to outweigh the gain of not needing to make a database query on each call.

I'll look into this a bit more and try to add it, unless you think the stale-cache risk is too great.